sklearn/metrics/pairwise.py

```diff
@@ -1041,6 +1303,7 @@ def distance_metrics():
     'l1'             metrics.pairwise.manhattan_distances
     'l2'             metrics.pairwise.euclidean_distances
     'manhattan'      metrics.pairwise.manhattan_distances
+    'gower'          metrics.pairwise.gower_distances
```

Comment:
@marcelobeckmann you have add `gower` to the list of metric function, but `gower` has not been added to the `PAIRWISE_DISTANCE_FUNCTIONS` collection.

Reply:
Thanks @darena-mdsol, I'll have a look in the PAIRWISE_DISTANCE_COLLECTION. About the weights, it was a misunderstanding from the original Gower formula, that weight is not necessary, and won't be added in the future.

Reply:
@marcelobeckmann Thanks for your reply. I'm an on a project that allows a user to assign weight/importance to each variable. My team has been discussing if weight/importance is useful or not, in lieu of your reply. Could you explain the misunderstanding that you mentioned? Thanks.

Reply:
Hi darena-mdsol, the purpose of this PR is to implement the default Gower distance as described in the section 1 from original paper. If the weighted implementation as described in the section 4 needs to be implemented, then a new ticket needs to be open, I won't do that in this PR. My misunderstanding was the formulas in the section 4 were the main proposal from that paper.